llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7190,6 +7190,29 @@ Instruction *InstCombinerImpl::visitICmpInst(ICmpInst &I) {
       }
     }
 
+    // These transform works when C is negative.
+    // X s< X^C, X s<= X^C, X u> X^C, X u>= X^C  --> X s< 0
+    // X s> X^C, X s>= X^C, X u< X^C, X u<= X^C  --> X s>= 0
+    const APInt *C;
+    if (match(Op0, m_c_Xor(m_Specific(Op1), m_APInt(C))) && C->isNegative()) {
```

Comment:
Maybe I missed it somewhere, but it looks like while the implementation was generalized to isNegative(), all the tests still use -1 only? It would be good to change some of them to a different negative value.

Reply:
I'm not against it, but can you tell me why it should be? just for  growing knowledge about llvm.